plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -71,7 +71,7 @@ export const EntityRefLink = forwardRef<any, EntityRefLinkProps>(
     kind = kind.toLocaleLowerCase('en-US');
     namespace = namespace?.toLocaleLowerCase('en-US') ?? DEFAULT_NAMESPACE;
 
-    const routeParams = { kind, namespace, name };
```

Comment:
I guess in theory namespace could also have special characters? Kind should be more restricted

Reply:
Agreed - maybe we should just encode them all to be thorough, there's little reason not to.

That being said, I'm mildly surprised that the result of `useRouteRef` doesn't help with encoding. Was that really intentional? Here's the code that ultimately gets called when forming the URL https://github.com/backstage/backstage/blob/0babe765539fadbd1999b181efac03f5385c8b05/packages/core-app-api/src/routing/RouteResolver.ts#L183

@Rugvip

Reply:
I've encoded `kind` and `namespace` as suggested.

> I'm mildly surprised that the result of useRouteRef doesn't help with encoding

I'm not sure whether changing the implementation of `useRouteRef` might introduce any side effects 🤔
but happy to circle back on this in case you want to continue on this path